Contributor sign up form

Mariatta opened the pull request:
I'll create the google forms based on the feedback to this PR.

Mariatta committed:
Contributor sign up form

I'll create the google forms based on the feedback to this PR.

Mariatta commented on forms/contributor-signup.md:
```diff
@@ -0,0 +1,40 @@
+
+2. Your name (required)
+
+3. Are you a member of underrepresented group?
```
I'm not sure if this should be asked at all?

Nikoleta-v3 replied:
:thinking: I believe there is value in collecting information like that. The aim of the sprint is to introduce new contributors to open source but we do  want to mainly target underrepresented groups. And collecting these data will help us know whether we achieved our goal. 

I would regret, however, if a person of a represented group that does not feel comfortable attending the sprints not taking this opportunity to learn about open source because the felt intimidated by the question, or even people of underrepresented feeling offended (:woman_shrugging: ). 

For these reasons I would suggest a small modification:

```
- questions 3 (text and not a required field): Preferred pronouns 
```
and later on like question 10:
```
- question 10: Do you consider yourself a member of an underrepresented group?
- yes
- no
- prefer not to say
```

what do you think?

trallard replied:
I generally prefer the wording that Nikoleta suggested (usually followed by an 'if so, please feel free to elaborate')


But since we are indeed having a main focus on underrepresented folks I wonder if this is then unnecessary

trallard replied:
That said it might be useful if we plan to do an impact report or something similar to what  the women in ML did for [https://reshamas.github.io/impact-report-for-wimlds-scikit-learn-sprints/](https://reshamas.github.io/impact-report-for-wimlds-scikit-learn-sprints/)

nnja replied:
A little off topic, but if someone writes their preferred pronouns, how will we keep track of that and match it to the person? If someone fills in their pronouns on the form for an event and isn't called by those pronouns at the event itself, that might not be the best experience.

trallard replied:
Following on the pronouns:
- PyCon provides pronoun ribbons for all the attendees, so if an attendee has a pronoun ribbon that covers it for us. However, if the person chooses not to use one or did not know there were ribbons available, shall we be prepared and have stickers/something with other options or pronouns? (i.e. nor pronoun just my name, ask me, etc).

- I suggest changing `- questions 3 (text and not a required field): Preferred pronouns` for `- questions 3 (text and not a required field): Used pronouns`

Mariatta commented on forms/contributor-signup.md:
```diff
@@ -0,0 +1,40 @@
+
+   Any skill level is welcome! We plan to have a git helpdesk.
+
+10. Do you have your own laptop for the sprint, or will you require a loaner?
```
Not yet sure if this is something we can accommodate, but figured it would be good to ask?

nnja replied:
I would change this message a little bit, just in case we can't have loaners so we don't get people's hopes up.

Maybe something like
> Will you be able to bring your own laptop to the sprint? If not, there is a small chance that we will have extra computers available for your use.

Is that too verbose?

trallard replied:
I like this suggestion even if verbose :)

Mariatta committed:
Update contributor-signup.md

trallard commented on forms/contributor-signup.md:
```diff
@@ -0,0 +1,40 @@
+
+3. Are you a member of underrepresented group?
+
+4. Preferred open source projects to contribute to (should this be required?)
```
I do not think this should be required, but could perhaps rephrase to something like:

4. Have you got some projects in mind you would like to contribute to?

trallard commented on forms/contributor-signup.md:
```diff
@@ -0,0 +1,40 @@
+
+10. Do you have your own laptop for the sprint, or will you require a loaner?
+
+11. Additional notes or comments to sprint organizer?
```
Can we add something like:

Do you have any accessibility needs we should know about?

Mariatta replied:
Yes! I will add this to the other form as well.

trallard commented:
Now, this might be coming from me being based in the UK but shall we add a note at the top of the form indicating what we are collecting the info for, who is going to have access to it and the such. 
I expect people from all over the globe to register so it is always better to be clear about this when collecting personal data

trallard commented:
Finally, I would also suggest having a checked box section similar to:

What Operating System do you normally use/are comfortable with?
- Windows
- Linux 
- Mac OSX

nnja commented:
Would this be the right place to ask participants to install pre-reqs? Something like, "would you be able to install the software necessary to participate, such as X, Y, and Z? Or would we send this information in a follow up email?

Have we even thought of what pre-reqs should be? Definitely git.exe for Windows. It may be useful to have Mac folks upgrade their gjt. Depending on is version the system git may be very out of date. Possibly Python 3?

trallard commented:
I think it would be best in a follow-up email with the minimum installs (updated git, command line, python3, text editor)

Also, we can ask them to check the installs for the projects that will be sprinting/mentoring on the day and try and get their install up?

Mariatta committed:
Update contributor-signup.md

Mariatta committed:
Update contributor-signup.md

Mariatta commented:
Thanks all, I've updated the doc based on all the feedback:

- changed question 3 to "Used pronouns" (optional texbox)

- rephrased question 4 to: Have you got some projects in mind you would like to contribute to? (it will be paragraph type)

- topic interest areas, I copied the choices from the other form (web development, scientific, ...)

- rephrased the question about laptop loaner to: "Will you be able to bring your own laptop to the sprint? If not, there is a small chance that we will have extra computers available for your use." (textbox)

- added question about accessibility needs

- added question about "Do you consider yourself a member of an underrepresented group?"

- added "If you're a member of an underrepresented group, please elaborate", if they said yes in the previous question

- added question about operating system

> shall we add a note at the top of the form indicating what we are collecting the info for, who is going to have access to it and the such.
I expect people from all over the globe to register so it is always better to be clear about this when collecting personal data

We have this for PyCascades, if it sounds good I can add it:

> One of the pride points of our community is the diversity of our members. In this optional question, we'd like to know your demographic information. Your personal information will not be shared with any third party for any reason, but we might publish anonymous statistics to help us improve our diversity metrics.

The form has been created at https://docs.google.com/forms/d/e/1FAIpQLSfmUDvXXHRgkBpYcHpy6O3txHEeLihz6g7hlXHThlLJMJ2zig/viewform?usp=sf_link however I have not filled in all the questions there.

I've added you all as collaborator to the form, if you didn't get the invite let me know, and I'll resend.

trallard commented:
I have added the following changes directly to the form:

- What is your level of familiarity with Git? (multiple choice)
  - I have never used git before
  - I know what version control and git are
  - I have used git in the past for personal projects
  - I have used git to work in projects with other people
  - I use git on a regular basis (i.e can do forks, pull requests and code reviews)

I think this will help us to better identify what their level actually is.

And the description at the top now reads:
```
Please fill in this form if you would like to be mentored during the sprint.

For more information, please check: https://us.pycon.org/2019/hatchery/mentoredsprints/


One of the pride points of our community is the diversity of our members. In this questionnaire we are asking for some demographic information. Your personal information will not be shared with any third party for any reason, but we might publish anonymous statistics to report the impact of this mentored sprints. If you have any questions or concerns about how your data might be used please contact the organisers at nwuk@pyladies.com

```

If we are happy with the forms I can ping Jackie and the hatchery committee and announce that we are opening the call. 

Also I want to double check that we are happy with this to be open until the 14th of March 2019 as we originally mentioned on the submission

Mariatta committed:
Update the question for git familiarity

trallard commented:
Merging this now, thanks a lot @Mariatta ✨
